dll/appcompat/apphelp/shimeng.c

```diff
@@ -207,6 +208,35 @@ BOOL WINAPIV SeiDbgPrint(SEI_LOG_LEVEL Level, PCSTR Function, PCSTR Format, ...)
     return TRUE;
 }
 
+static
+BOOL SeiIsOrdinalName(LPCSTR lpProcName)
+{
+    return (ULONG_PTR)lpProcName <= MAXUSHORT;
+}
+
+LPCSTR SeiPrintFunctionName(LPCSTR lpProcName, char szOrdProcFmt[10])
+{
+    if (SeiIsOrdinalName(lpProcName))
+    {
+        StringCchPrintfA(szOrdProcFmt, 10, "#%Iu", (ULONG_PTR)lpProcName);
+        return szOrdProcFmt;
+    }
+    return lpProcName;
+}
+
+int SeiCompareFunctionName(LPCSTR lpProcName1, LPCSTR lpProcName2)
+{
+    BOOL Ord1 = SeiIsOrdinalName(lpProcName1);
+    BOOL Ord2 = SeiIsOrdinalName(lpProcName2);
+
+    if (Ord1 != Ord2)
+        return 1;
+    if (Ord1)
+        return (ULONG_PTR)lpProcName1 == (ULONG_PTR)lpProcName2;
```

Comment:
Maybe you actually need to return the **difference** of these two pointers (casted to ULONG_PTR of course), since it seems that, based on the default case where you call strcmp(), your SeiCompareFunctionName() function will return 0 when both names are "equal", but will return a <0 or >0 value when one of the names is "before" or "after" the other one.